tls/s2n_client_cert_verify.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2014 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+
+#include "error/s2n_errno.h"
+#include "tls/s2n_client_extensions.h"
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_config.h"
+#include "tls/s2n_tls.h"
+
+#include "stuffer/s2n_stuffer.h"
+
+#include "utils/s2n_safety.h"
+
+
+int s2n_client_cert_verify_recv(struct s2n_connection *conn)
+{
+    struct s2n_stuffer *in = &conn->handshake.io;
+
+    s2n_hash_algorithm chosen_hash_alg = S2N_HASH_MD5_SHA1;
+    s2n_signature_algorithm chosen_signature_alg = S2N_SIGNATURE_RSA;
+
+    if(conn->actual_protocol_version == S2N_TLS12){
+        int pairs_available = 1;
+        /* Read and Validate Hash and Signature Algorithms provided by Client */
```

Comment:
I think comment should call out something like: "Make sure the client is actually using one of the {sig,hash} paris that we advertised in the `CertificateRequest`" . 
This satisfies 
```
The hash and signature algorithms used in the signature MUST
be one of those present in the supported_signature_algorithms
field of the CertificateRequest message. 
``` 
in https://tools.ietf.org/html/rfc5246#section-7.4.8

I was skimming through the code and missed the subtlety of what line 40 is accomplishing w.r.t. to the RFC.